src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -169,36 +187,79 @@ public void apply(VariantContext vc, ReadsContext reads, ReferenceContext ref, F
+        // For polymorphic sites we need to make sure e.g. the SB tag is sent to the annotation engine and then removed later.
+        // For monomorphic sites we need to make sure e.g. the hom ref genotypes are created and only then are passed to the annotation engine.
+        // We could theoretically make 2 passes to re-create the genotypes, but that gets extremely expensive with large sample sizes.
         if ( result.isMonomorphicInSamples() ) {
             if ( !includeNonVariants) {
                 return null;
             } else {
                 // For monomorphic sites we need to make sure e.g. the hom ref genotypes are created and only then are passed to the annotation engine.
                 final VariantContext reannotated = new VariantContextBuilder(result).genotypes(cleanupGenotypeAnnotations(result, true)).make();
-                return annotationEngine.annotateContext(reannotated, features, ref, null, a -> true);
+                return annotationEngine.annotateContext(reannotated, features, ref, null, GenotypeGVCFs::annotationShouldBeSkippedForHomRefSites);
             }
         } else {
             // For polymorphic sites we need to make sure e.g. the SB tag is sent to the annotation engine and then removed later.
             final VariantContext reannotated = annotationEngine.annotateContext(result, features, ref, null, a -> true);
             return new VariantContextBuilder(reannotated).genotypes(cleanupGenotypeAnnotations(reannotated, false)).make();
         }
+    }
+
+    private VariantContext calculateGenotypes(VariantContext vc){
+
+        final VariantContext.Type type = vc.getType();
+        final GenotypeLikelihoodsCalculationModel model;
+        /*
+         * Query the VariantContext for the appropriate model.  If type == MIXED, one would want to use model = BOTH.
+         * However GenotypingEngine.getAlleleFrequencyPriors throws an exception if you give it anything but a SNP or INDEL model.
+         */
+        if ( type == VariantContext.Type.INDEL ) {
+            model = GenotypeLikelihoodsCalculationModel.INDEL;
+        } else {
+            model = GenotypeLikelihoodsCalculationModel.SNP;
+        }
+        return genotypingEngine.calculateGenotypes(vc, model, null);
+    }
 
+    /**
+     * Determines whether the provided VariantContext has real alternate alleles
+     *
+     * @param vc  the VariantContext to evaluate
+     * @return true if it has proper alternate alleles, false otherwise
+     */
+    private boolean isProperlyPolymorphic(final VariantContext vc) {
+        return ( vc != null &&
+                !vc.getAlternateAlleles().isEmpty() &&
+                (!vc.isBiallelic() ||
+                        (!vc.getAlternateAllele(0).equals(Allele.SPAN_DEL) &&
+                                !vc.getAlternateAllele(0).equals(GATKVCFConstants.SPANNING_DELETION_SYMBOLIC_ALLELE_DEPRECATED))
```

Comment:
I don't think I removed that deliberately?  I'll add it in and see if it breaks anything...